Src/IronPython/Runtime/StringDictionaryStorage.cs

```diff
@@ -149,12 +150,17 @@ internal class StringDictionaryStorage : DictionaryStorage {
         public override bool HasNonStringAttributes() {
             if (_data != null) {
                 lock (this) {
-                    if (TryGetObjectDictionary() != null) {
-                        return true;
+                    Dictionary<object, object> dataDict = TryGetObjectDictionary();
```

Comment:
My only concern here is that different `DictionaryStorage` return different answers? For example `CommonDictionaryStorage` would return `true` if it contains `Extensible<string>` elements?

Reply:
Good point. I actually went though all concrete classes that override this method to make sure they are consistent (so no, `CommonDictionaryStorage` would return `false` if it contains only `string` or `Extensible<string>` elements), but forgot the abstract `DictionaryStorage` itself. Will fix it.